include/pldm_fw.hpp

```diff
@@ -81,12 +97,46 @@ class PldmFramework
 
     /**
      * @brief Get instance ID
-     * This api returns the instance id by making a dbus call to GetInstanceId
-     * api of Pldm. Instance id is to uniquely identify a message packet. This
-     * id is generated by pldm.
+     * Instance id is to uniquely identify a message packet. This api fetches
+     * the instance ID data base from libpldm and calls libpldm to allocate the
+     * valid instance ID.
+     *
+     * @return pldm_instance_id_t instance id.
+     *
+     * @throw Exception when failed to allocate instance ID.
+     */
+    pldm_instance_id_t getInstanceID();
+
+    /**
+     * @brief Free PLDM requester instance id
+     * This API frees up valid instance ID along side it takes care of
+     * destroying the instance ID data base as well.
+     *
+     * @param[in] instanceID - Instance ID.
+     */
+    void freeInstanceID(pldm_instance_id_t instanceID);
+
+    /**
+     * @brief setup PLDM transport for sending and receiving messages
+     *
+     * @return file descriptor of the socket established by PLDM for
+     * communication between BMC and host.
+     *
+     * @throw Exception when BMC to HOST(via PLDM) connection establishment
+     * fails.
+     */
+    int openPldm();
+
+    /**
+     * @brief Opens the MCTP socket for sending and receiving messages.
      *
-     * @return one byte instance id.
+     * @return file descriptor to perform MCTP demux based communication.
```

Comment:
just file descriptor of socket would be good

Reply:
done